### DISCUSSION TITLE
Reasoning behind enabled by default linters?

### DISCUSSION THREAD

**Discussion by jalaziz** [golangci/golangci-lint#2561] [Q&A] [answered]
Apologies if this has been answered elsewhere, but what is the reasoning behind the current set of default linters? In particular, why are `gofmt` and `goimports` not enabled by default?

We're starting to adopt `golangci-lint` and one thing that stood out was `gofmt` was not being enabled by default. After reading the docs, it was not clear to me why the default set was chosen.

**Comment by bombsimon** [comment, marked as answer]
The simple answer would be that it was decided by the author [many years back](https://github.com/golangci/golangci-lint/blob/034728ec9406111ceb90c2ee6015ccb4459a1ccc/pkg/lint/lintersdb/lintersdb.go#L177-L189) and has not been changed much to reduce amount of unexpected behaviour or breaking changes. At this point `golangci-lint` has so many linters and configurations it's hard to do a one-config-fits-all because we have such different needs and I think it's resonable to assume users adopting the tool will also configure it for their needs.

Similar discussions around sensible default has existed for things like [exclude-use-default](https://github.com/golangci/golangci-lint/issues/2239) which also shows that some defaults chosen for `golangci-lint` does not fit everyone.

**Reply by jalaziz** [reply]
Thanks for the quick reply.

Going [further back](https://github.com/golangci/golangci-lint/commit/30388862d25566de184411d61acdc23e49c54445#diff-f89262da2f86dbec85b3d9993ecd20d66fac45a29c92ef41165a2d221997d78eR35-R42) it seems that `golint` was being used as a replacement for `gofmt` and `goimports`. It was then later [disabled by default](https://github.com/golangci/golangci-lint/commit/d993d3a264f2cdfc179e196d5eeb95d3d9b6ad23#diff-d370a82fb385f97b1397cbef03b2a5c866b2bf9467a25b447d8a645c0d0c86b2R91-R109), but there's not much context as to why.

Thanks for pointing out the `exclude-use-default` issue. Seems like the defaults are pretty specific to the original needs and not as community standard as we thought. Still a great project though!
